server/routes.js

```diff
@@ -0,0 +1,254 @@
+							admin: true
+						}
+					}
+				})
+	},
+	{
+		path:/^\/tag\/([a-z0-9\-]{1,16})$/,
+		page: "page.hbs",
+		cache: true,
+		prerender: (db) => {
+			return db.posts.distinct("tags", {})
+				.then((tags) => tags.map((tag) => `/tag/${tag}`))
+		},
+		context: ([_, tag], db) =>
+			aggregatePosts(db, { $match: { tags: tag } }, { $sort: { timestamp: -1 } })
+				.then(fillAuthorInfo(db))
+				.then((posts) => {
+					posts.forEach((post) => post.short = true);
+					return {
+						title: `Tagged "${tag}"`,
+						posts: posts
+					};
+				})
+				.then(fillDefaultSidebar(db))
+	},
+	{
+		path:/^\/author\/(matt|zach|ben|danial|david|ellis|kyle|thomas)$/, // sigh...
+		page: "page.hbs",
+		cache: true,
+		prerender: Object.keys(config.adminInfo).map((author) => `/author/${author}`),
+		context: ([_, author], db) =>
+			aggregatePosts(db, { $match: { author: users[author].gid } }, { $sort: { timestamp: -1 } })
+				.then(fillAuthorInfo(db))
+				.then((posts) => {
+					posts.forEach((post) => post.short = true);
+					return {
+						title: `Posts by ${author}`,
+						posts: posts,
+						filter: {
+							type: "author",
+							author: users[author]
+						}
+					};
+				})
+				.then(fillDefaultSidebar(db))
+	},
+	{
+		path:/^\/posts\/([0-9a-zA-Z_-]{7,14})$/,
+		page: "page.hbs",
+		cache: true,
+		prerender: (db) => {
+			return db.posts.find({})
+				.then((posts) => posts.map((post) => `/posts/${post.guid}`));
+		},
+		context: ([_, postId], db) =>
+			db.posts.findOne({ guid: postId })
+				.then(fillAuthorInfo(db))
+				.then((post) => ({
+						title: post.title.text,
+						posts: [post]
+					}))
+	},
+	{
+		path:/^\/preview\/([0-9a-zA-Z_-]{7,14})$/,
+		page: "page.hbs",
+		cache: false,
+		context: ([_, postId], db) =>
+			db.posts.findOne({ guid: postId })
+				.then(fillAuthorInfo(db))
+				.then((post) => {
+					post.hideComments = true;
+					return {
+						title: post.title.text,
+						posts: [post]
+					};
+				})
+	},
+	{
+		path:/^\/raw\/([0-9a-zA-Z_-]{7,14})$/,
+		page: "raw.hbs",
+		cache: true,
+		prerenderFIXME: (db) => {
+			return db.posts.find({})
+				.then((posts) => posts.map((post) => `/raw/${post.guid}`));
+		}
+	},
+	{
+		path:/^\/admin\/?$/,
+		page: "admin.hbs",
+		cache: false,
+		context: (_, db) =>
+			aggregatePosts(db, { $sort: { timestamp: -1 } })
+				.then(fillAuthorInfo(db))
+				.then((posts) => ({
+					posts,
+					admin: true
+				}))
+	},
+	{
+		path:/^\/rss\/?$/,
+		page: "rss.hbs",
+		cache: true,
+		prerender: ["/rss"]
+	},
+	{
+		path:/^\/contact\/?$/,
+		page: "contact.hbs",
+		cache: true,
+		prerender: ["/contact"]
+	},
+	{
+		path:/^\/about\/?$/,
+		page: "about.hbs",
+		cache: true,
+		prerender: ["/about"]
+	},
+	{
+		path:/^\/404$/,
+		page: "not-found.hbs",
+		cache: true,
+		prerender: ["/404"],
+		context: {
+			fourohfour: true
+		}
+	},
+	{
+		path:/^\/manifest.json$/,
+		page: "manifest.json",
+		cache: true,
+		prerender: ["/manifest.json"]
+	}
+];
+
```

Comment:
Sure, but with the exception of `range` none of these could reasonably used anywhere outside of routes.